fastn-core/src/auth/routes.rs

```diff
@@ -11,19 +13,32 @@ pub async fn login(
 
     match provider.as_str() {
         "github" => fastn_core::auth::github::login(req, next).await,
+        // client should handle redirects to next for emailpassword login
+        "emailpassword" => fastn_core::auth::emailpassword::login(req, next).await,
         _ => Ok(fastn_core::not_found!("unknown provider: {}", provider)),
     }
 }
 
 // route: /-/auth/logout/
-pub fn logout(
+pub async fn logout(
     req: &fastn_core::http::Request,
     next: String,
 ) -> fastn_core::Result<fastn_core::http::Response> {
     // TODO: Refactor, Not happy with this code, too much of repetition of similar code
+
+    if let Some(session_id) = req.cookie(fastn_core::auth::COOKIE_NAME) {
+        let session_id =
+            uuid::Uuid::from_str(session_id.as_str()).expect("cookie contains valid uuid");
```

Comment:
We can not crash request (and wake up the entire developer team if some random user sends a request with malformed cookie). Just internalise every unwrap/expect means some developer have to wake up at 3AM and check the production log about why the site is giving server errors.